src/__tests__/fixtures/regression.css

```diff
@@ -0,0 +1,7 @@
+@custom-selector :--button .button;
+
+:--button {
+  &--active {
```

Comment:
Sorry but specs explicitly said that & is equivalent to :matches() https://tabatkins.github.io/specs/css-nesting/#selectordef-
so this just can't work...